File: app/src/main/java/com/games/memorygame/persistence/ScoreLocalDataSource.java
package com.games.memorygame.persistence;


import com.games.memorygame.model.Score;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Flowable;

public class ScoreLocalDataSource implements ScoreDataSource {

    private ScoreDao scoreDao;

    @Inject
    public ScoreLocalDataSource(ScoreDao scoreDao) {
        this.scoreDao = scoreDao;
    }

    @Override
    public Flowable<List<Score>> loadScores() {
        return scoreDao.getAllScores();
    }

    @Override
    public void addScore(Score score) {
        scoreDao.insertScore(score);
    }

    @Override
    public void clearData() {
        scoreDao.deleteAllScores();
    }
}

File: app/src/main/java/com/games/memorygame/persistence/ScoreLocalDataSource.kt
package com.games.memorygame.persistence


import com.games.memorygame.model.Score

import javax.inject.Inject

import io.reactivex.Flowable

class ScoreLocalDataSource @Inject
constructor(private val scoreDao: ScoreDao) : ScoreDataSource {

    override fun loadScores(): Flowable<List<Score>> = scoreDao.allScores

    override fun addScore(score: Score) {
        scoreDao.insertScore(score)
    }

    override fun clearData() {
        scoreDao.deleteAllScores()
    }
}